File: ncRegras/rCobranca.vb
Imports ncComum.nsExcecao
Imports ncDados.nsCobranca
Imports ncPersistencia
Imports ncPersistencia.nsCrediario
Namespace nsCobranca
    Public Class rCobranca

        Public Function ConsultarCobrancas(ByVal dados As dCobrancaAutomatica) As ColecaoCobranca

            Dim retorno As ColecaoCobranca
            Dim persistencia As pCobranca
            Dim retornoPersistencia As ColecaoCobranca

            Try

                retorno = New ColecaoCobranca
                persistencia = New pCobranca
                retornoPersistencia = persistencia.ListarCobrancas(dados)

                If Not retornoPersistencia Is Nothing Then
                    If retornoPersistencia.Count > 0 Then
                        retorno.AddRange(retornoPersistencia)
                    Else
                        retorno = Nothing
                    End If
                Else
                    retorno = Nothing
                End If

            Catch ex As Exception

                retorno = Nothing
                Throw New ExcecaoNascomercio("Erro em Consultar Crediario [" & Me.ToString() & "] - " & ex.Message)

            End Try

            Return retorno

        End Function
    End Class
End Namespace

File: Servicos/rCobranca.cs
using System;
using Comum;
using Repositorios;
using Modelos;

namespace Servicos.nsCobranca
{
    public class rCobranca
    {
        private readonly IpCobranca _repo;
        public rCobranca(IpCobranca repo) { _repo = repo; }

        public ColecaoCobranca ConsultarCobrancas(dCobrancaAutomatica dados)
        {
            try { return _repo.ListarCobrancas(dados); }
            catch (ExcecaoNascomercio) { throw; }
            catch (Exception ex) { throw new ExcecaoNascomercio("Erro em ConsultarCobrancas Cobranca [" + ToString() + "] - " + ex.Message); }
        }
    }
}
